app/src/services/guild.ts

```diff
@@ -0,0 +1,115 @@
+import { Contract, ethers, utils } from 'ethers'
+import { Web3Provider } from 'ethers/providers'
+import { BigNumber } from 'ethers/utils'
+
+import { getContractAddress } from '../util/networks'
+
+const GuildAbi = [
+  {
+    inputs: [{ internalType: 'uint256', name: 'amount', type: 'uint256' }],
+    name: 'lockTokens',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    inputs: [{ internalType: 'address', name: '', type: 'address' }],
+    name: 'tokensLocked',
+    outputs: [
+      { internalType: 'uint256', name: 'amount', type: 'uint256' },
+      { internalType: 'uint256', name: 'timestamp', type: 'uint256' },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'tokenVault',
+    outputs: [{ internalType: 'contract TokenVault', name: '', type: 'address' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'token',
+    outputs: [{ internalType: 'contract IERC20Upgradeable', name: '', type: 'address' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'lockTime',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'totalLocked',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [{ internalType: 'uint256', name: 'amount', type: 'uint256' }],
+    name: 'releaseTokens',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+]
+
+class OmenGuildService {
+  user: ethers.providers.JsonRpcSigner
+  network: number
+  provider: any
+  omenGuildAddress: string
+
+  constructor(provider: Web3Provider, network: number) {
+    const signer = provider.getSigner()
+    this.user = signer
+    this.network = network
+    this.provider = provider
+    this.omenGuildAddress = getContractAddress(network, 'omenGuildProxy')
+  }
+  get getOmenGuildContract(): Contract {
```

Comment:
Just a stylistic preference, but can you clean up the spacing between functions in this file?